### PR TITLE
New Goodie: NumPy Cheat Sheet

### PR DESCRIPTION
**Pull request by amarlearning**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New NumPy Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes
What changes does this PR introduce? 
New Goodie: NumPy Cheat Sheet

###### Related Issues or Discussions 
What issues or discussions (if any) does this PR address? Please provide the issue # or link.

###### People to notify
@mention any relevant people/organizations
@pjhampton 

---

Instant Answer Page: https://duck.co/ia/view/numpy_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=numpy cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/numpy_cheat_sheet_mobile.png?jNbVUDDqXk)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=numpy cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/numpy_cheat_sheet_mobile.png?sCpu6BwtxJ)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=numpy cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/numpy_cheat_sheet_mobile.png?qdTWMmagkj)

**Comment by amarlearning** [comment]
@tagawa plz review it!

File: share/goodie/cheat_sheets/json/numpy.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		"Load NumPy": [{
+			"key": "import numpy as np",
+			"val": "Conveniently imports the package with all the available NumPy modules."
+		}],
+		"Array Initialization": [{
+			"key": "np.array(\\[2, 3, 4\\])",
+			"val": "Direct initialization of array."
+		}, {
+			"key": "np.empty(20, dtype=np.float32)",
+			"val": "Single precision array of size 20."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: @amarlearning Can you please remove `.` from the end of every `val`?

File: share/goodie/cheat_sheets/json/numpy.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	"section_order": [
+		"Load NumPy",
+		"Array Initialization",
+		"Indexing",
+		"Array Properties and Operations",
+		"Elementwise Operations and Math Functions",
+		"Linear Algebra",
+		"Reading/ Writing Files",
+		"Rounding",
+        "Interpolation, Integration and Optimization"
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: @amarlearning Indentation issues here!

File: share/goodie/cheat_sheets/json/numpy.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+			"val": "Initializes 200 with value zero."
+		}, {
+			"key": "np.ones((3,3), dtype=np.int32)",
+			"val": "Creates a 3 x 3 integer matrix with ones init."
+		}, {
+			"key": "np.zeros_like(a)",
+			"val": "Creates an array with zeros and the shape of a."
+		}, {
+			"key": "np.copy(a)",
+			"val": "copy array to new memory."
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: @amarlearning For consistency, let's keep first letter of first word in every `val` capital. 🙂

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=numpy cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/numpy_cheat_sheet_mobile.png?qlj2pV9k5i)

**Comment by amarlearning** [comment]
@sahildua2305 plz review!

File: share/goodie/cheat_sheets/json/numpy.json [hunk 4]
```diff
@@ -0,0 +1,166 @@
+{
+	"id": "numpy_cheat_sheet",
+	"name": "numpy",
+	"description": "A reference to NumPy, the Python package for scientific computing.",
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: @amarlearning Remove the dot at the end of description please.

File: share/goodie/cheat_sheets/json/numpy.json [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+			"val": "Direct initialization of array"
+		}, {
+			"key": "np.empty(20, dtype=np.float32)",
+			"val": "Single precision array of size 20"
+		}, {
+			"key": "np.zeros(200)",
+			"val": "Initializes 200 with value zero"
+		}, {
+			"key": "np.ones((3,3), dtype=np.int32)",
+			"val": "Creates a 3 x 3 integer matrix with ones init"
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: I think you meant `in it`.

File: share/goodie/cheat_sheets/json/numpy.json [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+			"val": "Addition with array b"
+		}, {
+			"key": "np.exp(a)",
+			"val": "Exponential (complex and real)"
+		}, {
+			"key": "np.power(a, b)",
+			"val": "a to the power b"
+		}, {
+			"key": "np.sin(a)",
+			"val": "Returns sine of a, same goes for 'np.cos(a)'"
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: According to me, it's better to mention another entry for `np.cos(a)`.

File: share/goodie/cheat_sheets/json/numpy.json [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+			"key": "np.radians(a)",
+			"val": "Convert degrees to radians"
+		}, {
+			"key": "np.degrees(a)",
+			"val": "Convert radians to degrees"
+		}, {
+			"key": "np.var(a)",
+			"val": "Variance of array"
+		}],
+		"Linear Algebra": [{
```
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: I find this section too complicated to make it to library cheat sheet. May be it's just me. What do you say?

File: share/goodie/cheat_sheets/json/numpy.json (hunk 5 above)
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/numpy.json]: that `init` refers to `initialised`

File: share/goodie/cheat_sheets/json/numpy.json (hunk 7 above)
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/numpy.json]: second `key` & `value`, but according to me the first one should be there. It's not that complicated.

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: Cool! Let's keep it then 👍

File: share/goodie/cheat_sheets/json/numpy.json (hunk 5 above)
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/numpy.json]: i think `Creates a 3 x 3 integer matrix with ones initialised` would be better. what do you say. your call ?

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/numpy.json]: Let's go with even simpler - `Creates a 3 x 3 integer matrix with all ones`.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=numpy cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/numpy_cheat_sheet_mobile.png?lF0lExIVe9)

**Comment by sahildua2305** [comment]
Thanks @amarlearning for making improvements. Looks good to me!

**Comment by daxtheduck** [comment]
@amarlearning this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**